### PR TITLE
ledger-tool: Use error handling in blockstore command code

### PR DESCRIPTION
**Pull request by steviez**
#### Problem
There are lots of operations that could fail, including lots of the
Blockstore calls. The old code matched on Ok(_) or did unwrap()'s
which clutter the code and increase indentation.

#### Summary of Changes
This change wraps the entire command in a function that returns a
Result. The wrapper then does a single unwrap_or_else() and prints
any error message. Everywhere else is now free to use the ? operator

### CODE REVIEW

File: ledger-tool/src/blockstore.rs
Comment (CriesofCarrots) [inline ledger-tool/src/blockstore.rs]: Hmm, specific Error type maybe? Given how few non-`BlockstoreError` sites there are, it's worth considering whether we could just return `BlockstoreError` here.
If absolutely a no-go, it might be nice to introduce a `LedgerToolError` enum, that we could eventually expand to handle errors across the tool. What do you think?

File: ledger-tool/src/blockstore.rs
Comment (CriesofCarrots) [inline ledger-tool/src/blockstore.rs:825]: So far this is the first line I've spotted that wouldn't return a `BlockstoreError`

File: ledger-tool/src/blockstore.rs
Comment (CriesofCarrots) [inline ledger-tool/src/blockstore.rs:269]: I guess this one too (not a `BlockstoreError`)

File: ledger-tool/src/blockstore.rs
Comment (CriesofCarrots) [inline ledger-tool/src/blockstore.rs:45]: Return a `BlockstoreError`?

File: ledger-tool/src/blockstore.rs
Comment (CriesofCarrots) [inline ledger-tool/src/blockstore.rs:115]: Return a `BlockstoreError`?

**Review by CriesofCarrots — COMMENTED** [review]
Generally, this looks great. I just wonder if we could return more-specific errors and hoist `Box<dyn std::error::Error>`.

File: ledger-tool/src/blockstore.rs
Comment (steviez) [reply ledger-tool/src/blockstore.rs]: I like this idea; went ahead and added a `LedgerToolError`. Not much work given that `thiserror` does most of the heavy lifting

File: ledger-tool/src/blockstore.rs
Comment (steviez) [reply ledger-tool/src/blockstore.rs]: On this one, I just went ahead and went straight to returning a `Result<(), LedgerToolError`. Technically, this function can only return a `BlockstoreError` tho ... let me know if you think we should stick with `BlockstoreError` for longer

File: ledger-tool/src/blockstore.rs
Comment (steviez) [inline ledger-tool/src/blockstore.rs:604]: I think it'd be nicer if the subcommand functions all return a common `Result` type, and we can have a single `unwrap_or_else()` within `main.rs`.

However, I started working on updating the other commands, and hit some minor "snags" with `bigtable.rs`. Namely, `solana_ledger::bigtable_upload::upload_confirmed_blocks()` and `solana_ledger::bigtable_delete::delete_confirmed_blocks()` return a `Result<(), Box<dyn std::error::Error>`.

This was seemingly done as there are subcalls that can return a `BlockstoreError` or a `solana_storage_bigtable::Error`. Guessing the same dilemma is how it came to be that the bigtable ledger-tool commands return a `Result<(), Box<dyn std::error::Error>` as well.

Kind of the fence here - part of me wants to clean those up right now, but part of me is also thinking we have a decent logical boundary and could handle additional condensing in a followup. You leaning one way or the other ?

Comment (CriesofCarrots) [reply ledger-tool/src/blockstore.rs:604]: Yeah, I started looking at the `bigtable` subcommand as well. While I haven't really dug in yet, I like the idea of cleaning those up lower and being able to use the LedgerToolError there as well.

But I see no real drawback in merging this one as-is, and doing the additional condensing later.

File: ledger-tool/src/blockstore.rs
Comment (CriesofCarrots) [reply ledger-tool/src/blockstore.rs]: I have a slight preference for using the "lower" error as long as we can (ie. `analyze_column` and `analyze_storage` both return BlockstoreError), but also acknowledge it's not particularly ergonomic to have different Result types in the same module.
I'll leave it up to you. Approving, but happy to re-approve if you're on board with that change.

**Review by CriesofCarrots — APPROVED** [review]

File: ledger-tool/src/blockstore.rs
Comment (steviez) [reply ledger-tool/src/blockstore.rs]: I agree with the sentiment and started making the change. However, I was then reminded that `analyze_column()` (and thus `analyze_storage()`) can return a `serde_json::Error`. Namely, from this line:
```rust
println!("{}", serde_json::to_string_pretty(&json_result)?);
```
So, I can pretend I knew what I was doing when I first implemented this 😉 ; going to push the PR as-is

Comment (CriesofCarrots) [reply ledger-tool/src/blockstore.rs]: Perfect. Sorry for missing the serde_json error!

**Pull request merged**
